src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -541,4 +664,61 @@ private static String typeSwitchClassName(Class<?> targetClass) {
+            if (from == byte.class || from == short.class || from == char.class) {
+                from = int.class;
+            }
+            return new TypePairs(from, to);
+        }
+
+        public static Map<TypePairs, String> initialize() {
+            Map<TypePairs, String> typePairToName = new HashMap<>();
+            typePairToName.put(new TypePairs(byte.class,   char.class),   "isIntToCharExact");      // redirected
+            typePairToName.put(new TypePairs(short.class,  byte.class),   "isIntToByteExact");      // redirected
+            typePairToName.put(new TypePairs(short.class,  char.class),   "isIntToCharExact");      // redirected
+            typePairToName.put(new TypePairs(char.class,   byte.class),   "isIntToByteExact");      // redirected
+            typePairToName.put(new TypePairs(char.class,   short.class),  "isIntToShortExact");     // redirected
+            typePairToName.put(new TypePairs(int.class,    byte.class),   "isIntToByteExact");
+            typePairToName.put(new TypePairs(int.class,    short.class),  "isIntToShortExact");
+            typePairToName.put(new TypePairs(int.class,    char.class),   "isIntToCharExact");
+            typePairToName.put(new TypePairs(int.class,    float.class),  "isIntToFloatExact");
+            typePairToName.put(new TypePairs(long.class,   byte.class),   "isLongToByteExact");
+            typePairToName.put(new TypePairs(long.class,   short.class),  "isLongToShortExact");
+            typePairToName.put(new TypePairs(long.class,   char.class),   "isLongToCharExact");
+            typePairToName.put(new TypePairs(long.class,   int.class),    "isLongToIntExact");
+            typePairToName.put(new TypePairs(long.class,   float.class),  "isLongToFloatExact");
+            typePairToName.put(new TypePairs(long.class,   double.class), "isLongToDoubleExact");
+            typePairToName.put(new TypePairs(float.class,  byte.class),   "isFloatToByteExact");
+            typePairToName.put(new TypePairs(float.class,  short.class),  "isFloatToShortExact");
+            typePairToName.put(new TypePairs(float.class,  char.class),   "isFloatToCharExact");
+            typePairToName.put(new TypePairs(float.class,  int.class),    "isFloatToIntExact");
+            typePairToName.put(new TypePairs(float.class,  long.class),   "isFloatToLongExact");
+            typePairToName.put(new TypePairs(double.class, byte.class),   "isDoubleToByteExact");
+            typePairToName.put(new TypePairs(double.class, short.class),  "isDoubleToShortExact");
+            typePairToName.put(new TypePairs(double.class, char.class),   "isDoubleToCharExact");
+            typePairToName.put(new TypePairs(double.class, int.class),    "isDoubleToIntExact");
+            typePairToName.put(new TypePairs(double.class, long.class),   "isDoubleToLongExact");
+            typePairToName.put(new TypePairs(double.class, float.class),  "isDoubleToFloatExact");
+            return typePairToName;
```

Comment:
This should return an immutable map, so that `TypePairs` lookups can be treated as constant:
```suggestion
            return Map.copyOf(typePairToName);
```